### ISSUE TITLE
Bot inventories and tools

### ISSUE THREAD

**Issue by tiansh** [JoeStrout/Farmtronics#8]
1. Bot can be picked up by hitting it with any tools. And after place the bot onto farm again, inventories of the bot are been reset: Anything the bot previous hold will be lost.
2. Player may also move water can or other tools from inventories of the bot to themselves inventories or move player's tools to the bot's inventories. If so, after pick up the bot and place it again, player may get as many tools as they want / lost their tools.
3. If the Bot is destroyed by an villager (when it been placed on their walking pathway), its inventories will also be lost (not be dropped).

![Game screenshot](https://user-images.githubusercontent.com/6500326/151993853-503cfc56-245e-4c60-ada4-e7fbfc619177.png)

Chest in the game cannot be picked up if it contains anything. And if it been destroyed by an villager, its contains will be dropped. Maybe this behavior would be more suitable if possible.

**Comment by JoeStrout** [comment]
Yes.  This too is in the known issues in the roadmap: https://github.com/JoeStrout/Farmtronics/blob/main/ROADMAP.md

But since some people will certainly be looking in Issues rather than at the roadmap, it's good to have it here as well.  Thank you for reporting it.

**Comment by shinysocks** [comment]
When a player's tool is missing from their inventory and instead in a bot, ex. having two watering cans in a bot, the game thinks the item was lost and gives you another in the lost and found box at Mayor Lewis' house.

**Comment by JoeStrout** [comment]
I don't think this reset-inventory issue happens anymore.  Can anyone still reproduce it?

**Comment by TSRBerry** [comment]
1. should also be solved with #35

I'll try to do something about 3. as well.

**Comment by JoeStrout** [comment]
This should've been fixed as of v1.0.7 of the mod.  After this next big release, we need to go through and close stale issues!
